operator/validator/controller.go

```diff
@@ -1242,38 +1256,31 @@ func (c *controller) startCommittee(vc *validator.Committee) (bool, error) {
 	return true, nil
 }
 
-func (c *controller) ForkMonitor(logger *zap.Logger) {
+func (c *controller) ForkListener(logger *zap.Logger) {
 	if c.networkConfig.PastAlanFork() {
 		return
 	}
 
-	bn := c.beacon.GetBeaconNetwork()
-
-	forkTime := time.Unix(bn.EstimatedTimeAtSlot(bn.FirstSlotAtEpoch(c.networkConfig.AlanForkEpoch)), 0)
-	timeUntilFork := time.Until(forkTime)
+	slotTicker := slotticker.New(c.logger, slotticker.Config{
+		SlotDuration: c.networkConfig.SlotDurationSec(),
+		GenesisTime:  c.networkConfig.GetGenesisTime(),
+	})
 
+	next := slotTicker.Next()
 	for {
 		select {
-		case <-c.context.Done():
-			return
-		case <-time.After(timeUntilFork):
-			logger.Info("Stopping genesis validators after alan fork")
-			c.CleanGenesisValidators()
-			logger.Info("✅Successfully stopped genesis validators")
+		case <-c.ctx.Done():
 			return
+		case <-next:
+			next = slotTicker.Next()
```

Comment:
@moshe-blox, do we really need this?

Reply:
we don't really, but i don't trust very long sleeps to be precise

when we release Alan, the fork epoch could be set to around 2 weeks ahead, and i can't trust that if we schedule a Go timer for 2 weeks from now, it would fire precisely at the exact millisecond we need it to

given that system clocks often correct themselves, in such a long timeframe, i don't know what would be the impact of that